### PR TITLE
Integrate C-IL for multi-label classification

### PR DESCRIPTION
**Pull request by JihwanEom**
### This PR includes :
- Integrate class-incremental learning for multi-label classification
- Improve stability in stage.py

### CODE REVIEW

File: mpa/cls/stage.py [hunk 1]
```diff
@@ -74,7 +74,10 @@ def configure(self, model_cfg, model_ckpt, data_cfg, training=True, **kwargs):
                 cfg.data.num_classes = len(cfg.data.train.get('classes', []))
             cfg.model.head.num_classes = cfg.data.num_classes
 
-        if isinstance(cfg.model.head.topk, tuple):
+        if 'Multi' in cfg.model.head.get('type', False) and cfg.model.head.get('topk', False):
```
Comment (harimkang) [inline mpa/cls/stage.py]: Can't we use cfg.task as the standard (not head.type)?
I think we can also combine this if statement with 79-80 line and use it as one if statement.

File: models/classification/ote_efficientnet_b0_multilabel.yaml [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+model:
+    type: SAMImageClassifier
+    task: classification
+    backbone:
+        version: b0
+    head:
+        type: MultiLabelLinearClsHead
+        loss:
+            type: AsymmetricLoss
```
Comment (harimkang) [inline models/classification/ote_efficientnet_b0_multilabel.yaml]: need to add new line

File: mpa/cls/stage.py [hunk 3]
```diff
@@ -196,12 +199,14 @@ def configure_task(cfg, training, model_meta=None, **kwargs):
...
                 cfg.model.head.num_classes = len(dst_classes)
                 gamma = 2 if cfg['task_adapt'].get('efficient_mode', False) else 3
-                cfg.model.head.loss = ConfigDict(
-                    type='SoftmaxFocalLoss',
-                    loss_weight=1.0,
-                    gamma=gamma,
-                    reduction='none',
-                )
+
+                if 'Multi' not in cfg.model.head.type:
```
Comment (harimkang) [inline mpa/cls/stage.py:203]: As in the previous comment, we can use (head.type ->) cfg.task as the standard task type.

**Review by harimkang — CHANGES_REQUESTED** [review]

File: mpa/cls/stage.py (hunk 1 above)
Comment (harimkang) [reply mpa/cls/stage.py]: sorry, I was mistaken. This is using the same cfg.task as 'classification'.
I think you can change cfg.task to multi-label classification from classification 
or The conditional statement {'Multi' in cfg.model.head.get('type') }needs to be reconsidered. (need to be more specific.)

File: mpa/cls/stage.py (hunk 3 above)
Comment (harimkang) [inline mpa/cls/stage.py]: Same as above comment.

**Review by goodsong81 — APPROVED** [review]
LGTM. Please resolve Harim's comments.

File: mpa/cls/stage.py [hunk 4]
```diff
@@ -196,12 +199,14 @@ def configure_task(cfg, training, model_meta=None, **kwargs):
...
                 cfg.model.head.num_classes = len(dst_classes)
                 gamma = 2 if cfg['task_adapt'].get('efficient_mode', True) else 3
-                cfg.model.head.loss = ConfigDict(
-                    type='SoftmaxFocalLoss',
-                    loss_weight=1.0,
-                    gamma=gamma,
-                    reduction='none',
-                )
+
+                if not cfg.model.get('multilabel', False):
```
Comment (harimkang) [inline mpa/cls/stage.py:201]: 'If not' doesn't look good, do you have any other ideas?

File: mpa/cls/stage.py [hunk 5]
```diff
@@ -74,8 +74,11 @@ def configure(self, model_cfg, model_ckpt, data_cfg, training=True, **kwargs):
                 cfg.data.num_classes = len(cfg.data.train.get('classes', []))
             cfg.model.head.num_classes = cfg.data.num_classes
 
-        if isinstance(cfg.model.head.topk, tuple):
-            cfg.model.head.topk = (1,) if cfg.model.head.num_classes < 5 else (1, 5)
+        if cfg.model.head.get('topk', False) and isinstance(cfg.model.head.topk, tuple):
+            if cfg.model.get('multilabel', False):
```
Comment (harimkang) [inline mpa/cls/stage.py:79]: how about
```
cfg.model.head.topk = (1,) if cfg.model.head.num_classes < 5 else (1, 5)
if cfg.model.get('multilabel', False): cfg.model.head.pop('topk', None)
```

**Review by harimkang — CHANGES_REQUESTED** [review]
I left a few comments.

File: mpa/cls/stage.py (hunk 5 above)
Comment (JihwanEom) [reply mpa/cls/stage.py:79]: How do you think about below code?

```
cfg.model.head.topk = (1, cfg.model.head.num_classes) if cfg.model.head.num_classes < 5 else (1, 5)
if cfg.model.get('multilabel', False): cfg.model.head.pop('topk', None)
```

File: mpa/modules/models/classifiers/sam_classifier.py [hunk 6]
```diff
@@ -30,6 +30,8 @@ class SAMImageClassifier(ImageClassifier):
     """SAM-enabled ImageClassifier"""
 
     def __init__(self, task_adapt=None, **kwargs):
+        if 'multilabel' in kwargs:
```
Comment (JihwanEom) [inline mpa/modules/models/classifiers/sam_classifier.py:33]: @harimkang this multilabel flag is needed only for state_dict_hook. How do you think about separating state_dict_hook into two cases: 1) backbone, 2) head
If we separate this, each classifier will not have to remove multilabel argument. But I think state_dict hook in heads might be implemented.

**Review by harimkang — APPROVED** [review]
LGTM.
Need to check that there are no conflicts with recently merged code.

**Review by supersoob — APPROVED** [review]
LGTM
